### DISCUSSION TITLE
Single Instance

### DISCUSSION THREAD

**Discussion by myokeeh** [microsoft/TemplateStudio#4615] [Q&A] [answered]
One key difference I just noticed with my UWP to WinUI conversion is the WinUI version can run multiple instances--UWP app does not. What is the suggested approach to keep it from running multiple instances?

I think multiple instances or windows will be something I'll want eventually, but because the app deals with a SQLite database, I only want a single instance for now.

Can something like be added as boilerplate/option to Template Studio?

**Comment by mrlacey** [comment, marked as answer]
@myokeeh As a workaround to doing this yourself, see https://blogs.windows.com/windowsdeveloper/2022/01/28/making-the-app-single-instanced-part-3/

**Comment by mikebattista** [comment]
See https://learn.microsoft.com/en-us/windows/apps/windows-app-sdk/applifecycle/applifecycle-instancing#how-the-windows-app-sdk-instancing-differs-from-uwp-instancing for some guidance on this.

**Comment by mikebattista** [comment]
Duplicate of https://github.com/microsoft/TemplateStudio/issues/4662
